pennylane/capture/primitives.py

```diff
@@ -163,6 +163,31 @@ def __hash__(self):
     return AbstractMeasurement
 
 
+@lru_cache
+def create_non_jvp_primitive():
+    """Create a primitive type ``NonJVPPrimitive``, which binds to JAX's JVPTrace
+    like a standard Python function and otherwise behaves like jax.core.Primitive.
+    """
+
+    if not has_jax:  # pragma: no cover
+        return None
+
+    # pylint: disable=too-few-public-methods
+    class NonJVPPrimitive(jax.core.Primitive):
+        """A subclass to JAX's Primitive that works like a Python function
+        when evaluating JVPTracers."""
+
+        def bind_with_trace(self, trace, args, params):
+            """Bind the ``NonJVPPrimitive`` with a trace. If the trace is a ``JVPTrace``,
+            binding falls back to a standard Python function call. Otherwise, the
+            bind call of JAX's standard Primitive is used."""
+            if isinstance(trace, jax.interpreters.ad.JVPTrace):
+                return self.impl(*args, **params)
+            return super().bind_with_trace(trace, args, params)
+
+    return NonJVPPrimitive
```

Comment:
We want this new class of primitive for operators, which is handled like a standard Python function by `JVPTrace`. This is because the differentiation of PennyLane operators does _not_ happen through them returning a modified operator or an "OperatorDerivative" type, but simply by storing the `JVPTracer` in the created operator instance.
This way, the differentiation of the QNode via `jax.jvp` can go its standard path.